src/tests/ftest/util/soak_test_base.py

```diff
@@ -284,6 +289,123 @@ def harasser_job_done(self, args):
         self.harasser_results[args["name"]] = args["status"]
         self.harasser_args[args["name"]] = args["vars"]
 
+    def schedule_jobs(self, node_list):
+        """Schedule jobs with internal scheduler.
+
+        Args:
+            node_list (list): list of nodes to use in jobs
+        """
+        debug_logging(self.log, self.enable_debug_msg, "DBG: schedule_jobs ENTERED ")
+        job_queue = multiprocessing.Queue()
+        jobid_list = []
+        jobs_not_done = []
+        # remove any nodes marked as DOWN
+        node_list.difference_update(self.down_nodes)
+        path = os.getenv("PATH")
+        env = f"export PATH={path}"
+        lib_path = os.getenv("LD_LIBRARY_PATH")
+        if lib_path:
+            env = ";".join([env, f"export LD_LIBRARY_PATH={lib_path}"])
+        v_env = os.getenv("VIRTUAL_ENV")
+        if v_env:
+            env = ";".join([env, f"export VIRTUAL_ENV={v_env}"])
+        for job_dict in self.joblist:
+            jobid_list.append(job_dict["jobid"])
+            jobs_not_done.append(job_dict["jobid"])
+        self.log.info("Submitting %s jobs at %s", str(len(jobid_list)), time.ctime())
+        job_threads = []
+        while True:
+            if time.time() > self.end_time or len(jobs_not_done) == 0:
+                break
+            job_results = {}
+            # verify that there are enough nodes to run remaining jobs
+            if len(job_threads) == 0:
+                for job_dict in self.joblist:
+                    job_id = job_dict["jobid"]
+                    if job_id in jobs_not_done:
+                        node_count = job_dict["nodesperjob"]
+                        if len(node_list) < node_count:
+                            # cancel job
+                            self.soak_results.update({job_id: "CANCELLED"})
+                            self.log.info(
+                                "FINAL STATE: soak job %s completed with : %s at %s",
+                                job_id,
+                                "CANCELLED",
+                                time.ctime())
+                            jobs_not_done.remove(job_id)
+            for job_dict in self.joblist:
+                job_id = job_dict["jobid"]
+                if job_id in jobid_list:
+                    node_count = job_dict["nodesperjob"]
+                    if len(node_list) >= node_count:
+                        debug_logging(
+                            self.log, self.enable_debug_msg, f"DBG: node_count {node_count}")
+                        debug_logging(
+                            self.log,
+                            self.enable_debug_msg,
+                            f"DBG: node_list initial/queue {node_list}")
+                        job_node_list = node_list[:node_count]
+                        debug_logging(
+                            self.log,
+                            self.enable_debug_msg,
+                            f"DBG: node_list before launch_job {node_list}")
```

Comment:
Is this supposed to be logging `job_node_list`? Otherwise it doesn't seem to add anything different than the previous debug message.